pkgs/servers/mir/default.nix

```diff
@@ -0,0 +1,121 @@
+{ lib
+, pkgs
+, stdenv
+, fetchFromGitHub
+, cmake
+, extra-cmake-modules
+, boost
+, egl-wayland
+, libglvnd
+, glm
+, glog
+, lttng-ust
+, udev
+, glib
+, wayland
+, libxcb
+, xorg
+, libdrm
+, mesa
+, libepoxy
+, nettle
+, libxkbcommon
+, libinput
+, libxmlxx
+, libuuid
+, pcre2
+, freetype
+, libyamlcpp
+, python3Packages
+, libevdev
+, libselinux
+, libsepol
+, pcre
+, eglexternalplatform
+, doxygen
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mir";
+  version = "2.10.0";
+
+  src = fetchFromGitHub {
+    owner = "MirServer";
+    repo = "mir";
+    rev = "v${version}";
+    sha256 = "sha256-HPbToo7lhsHyx0w4oO2wp3zZ0xpr6SlEm2oE6Rm8b3E=";
+  };
+
+  postPatch = ''
+    # Fix error broken path found in pc file ("//")
+    for f in $(find . -name '*.pc.in') ; do
+      substituteInPlace $f \
+        --replace "/@CMAKE_INSTALL_LIBDIR@" "@CMAKE_INSTALL_LIBDIR@" \
+        --replace "/@CMAKE_INSTALL_INCLUDEDIR@" "@CMAKE_INSTALL_INCLUDEDIR@"
+    done
+
+    # Fix paths for generating drm-formats
+    substituteInPlace src/platform/graphics/CMakeLists.txt \
+      --replace "/usr/include/drm/drm_fourcc.h" "${libdrm.dev}/include/libdrm/drm_fourcc.h" \
+      --replace "/usr/include/libdrm/drm_fourcc.h" "${libdrm.dev}/include/libdrm/drm_fourcc.h"
+
+    # Fix server-platform install path
+    substituteInPlace src/platforms/CMakeLists.txt \
+      --replace "\''${CMAKE_INSTALL_PREFIX}/\''${CMAKE_INSTALL_LIBDIR}/mir/server-platform" "\''${CMAKE_INSTALL_LIBDIR}/mir/server-platform"
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    doxygen
```

Comment:
```
-- doxygen 1.9.5 (>= 1.8.0) available - enabling make target doc
```

Doesn't seem enough to actually build & install any documentation, you may have to play with either the `makeFlags` to build & install all normal targets + docs or add some `postBuild` & `postInstall` code to do it.

Reply:
`libxslt` in `nativeBuildInputs` and
```nix
buildFlags = [ "all" "doc" ];
```

will build & install the documentation.

![Bildschirmfoto von 2022-12-27 16-54-56](https://user-images.githubusercontent.com/23431373/209690520-37825291-69ab-435e-94a1-c1b882f140fa.png)